balkhash/postgres.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import datetime
+
+
+from sqlalchemy import Column, DateTime, String, UniqueConstraint
+from sqlalchemy import Table, MetaData
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy import create_engine
+from sqlalchemy.dialects import postgresql
+from sqlalchemy import and_
+
+from balkhash import settings
+from balkhash.dataset import Dataset, Bulk
+
+
+log = logging.getLogger(__name__)
+
+
+class PostgresDataset(Dataset):
+
+    def __init__(self, name):
+        super(PostgresDataset, self).__init__(name)
+        self.engine = create_engine(settings.DATABASE_URI)
+        self.table_name = name
+        meta = MetaData(self.engine)
+        meta.reflect()
+        self.table = Table(
+            name, meta,
+            Column('id', String(60)),
+            # We have to cast null fragment values to "" to make the
+            # UniqueConstraint work
+            Column('fragment', String(60), nullable=False, default=""),
+            Column('properties', postgresql.JSONB),
+            Column('schema', String(32)),
+            Column('timestamp', DateTime, default=datetime.datetime.utcnow),
+            UniqueConstraint('id', 'fragment'),
+            extend_existing=True
+        )
+        meta.create_all(self.engine)
+
+    def delete(self, entity_id=None, fragment=None):
+        with self.engine.begin() as conn:
+            table = self.table
+            if entity_id:
+                if fragment:
+                    statement = table.delete(and_(
+                        table.c.id == entity_id,
+                        table.c.fragment == fragment
+                    ))
+                else:
+                    statement = table.delete(table.c.id == entity_id)
+            conn.execute(statement)
```

Comment:
I think there's a bug here: if entity_id and fragment are None, `statement` will not be defined. In general, one thing I've always enjoyed about sqlalchemy is call chaining like this: 

```python
statement = table.delete()
if entity_id not None:
    statement = statement.where(table.c.id == entity_id)
if fragment not None:
    statement = statement.where(table.c.fragment == fragment)
conn.execute(statement)
```